tools/Tools.Common/Models/ParameterMetadata.cs

```diff
@@ -62,5 +62,41 @@ public class ParameterMetadata
         /// Specifies if the parameter has the ValidateNotNullOrEmpty attribute
         /// </summary>
         public bool ValidateNotNullOrEmpty { get; set; }
+
+        /// <summary>
+        /// Checks if two ParameterMetadata objects are equal by comparing
+        /// each of the properties.
+        /// </summary>
+        /// <param name="other">The ParameterMetadata object being compared to this object.</param>
+        /// <returns>True if the two objects are equal, false otherwise.</returns>
+        public override bool Equals(Object obj)
+        {
+            var other = obj as ParameterMetadata;
+            if (other == null)
+            {
+                return false;
+            }
+
+            var flag = true;
+
```

Comment:
clean up the vertical spaces in this file - no reason for veritcal spaces except after ending brackets